pkg/primitives/vm/utils.go

```diff
@@ -156,6 +156,27 @@ func (p *Manager) prepContainer(
 		machine.Entrypoint = "/corex --ipv6 -d 7 --interface eth0"
 	}
 
+	// clean up host keys
+	firstBootVolumeName := fmt.Sprintf("firstBoot:%s", wl.ID.String())
```

Comment:
What is this ? why do u create a new volume ? 

Look @Eslam-Nawara i think you completely got this completely wrong

The code above, create a volume where the flist will be mounted. (line 106) above.
You can think that this storage volume will hold the files from the mounted flist. Good ? The function storage.VolumeCreate will create a new one if the volume does not exist already this happen on the first boot of the vm.

This means, if I detect that this volume didn't exist before this call, it means i am booting this vm for the first time hence i can safely decide to delete the files **from that volume** (after the mount call)

What you do here is that u create completely out of context a new volume that u call firstBootVolume !!! 
Then what u do is u still immediately delete all files from the **zos  host** not from the Volume

What you will u need to do eventually is to delete the files from the the 
```go
filepath.Join(volume.Path, "etc", "ssh", "ssh_host_*")
```